tests/codemods/test_with_threading_lock.py

```diff
@@ -86,3 +86,93 @@ def test_no_effect_multiple_with_clauses(self, tmpdir, klass):
     ...
 """
         self.run_and_assert(tmpdir, input_code, input_code)
+
+
+class TestThreadingNameResolution(BaseSemgrepCodemodTest):
+    codemod = WithThreadingLock
+
+    @pytest.mark.parametrize(
+        "input_code,expected_code",
+        [
+            (
+                """from threading import Lock
+lock = 1
+with Lock():
+    ...
+""",
+                """from threading import Lock
+lock = 1
+lock_1 = Lock()
+with lock_1:
+    ...
+""",
+            ),
+            (
+                """from threading import Lock
+from something import lock
+with Lock():
+    ...
+""",
+                """from threading import Lock
+from something import lock
+lock_1 = Lock()
+with lock_1:
+    ...
+""",
+            ),
+            (
+                """import threading
+lock = 1
+def f(l):
+    with threading.Lock():
+        return [lock_ for lock_ in l]
+""",
+                """import threading
+lock = 1
+def f(l):
+    lock_1 = threading.Lock()
+    with lock_1:
+        return [lock_ for lock_ in l]
+""",
+            ),
+            (
+                """import threading
+with threading.Lock():
+    int("1")
+with threading.Lock():
+    print()
+var = 1
+with threading.Lock():
+    print()
+""",
+                """import threading
+lock = threading.Lock()
+with lock:
+    int("1")
+lock_1 = threading.Lock()
+with lock_1:
+    print()
+var = 1
+lock_2 = threading.Lock()
+with lock_2:
+    print()
+""",
+            ),
+            (
+                """import threading
+with threading.Lock():
+    with threading.Lock():
+        print()
+""",
+                """import threading
+lock_1 = threading.Lock()
+with lock_1:
+    lock = threading.Lock()
+    with lock:
+        print()
```

Comment:
notice this test. libcst processes the internal with statement first, so our naming will be like this. Not a huuuge deal